python/sdk_e2e/test_track_1/client_fixtures.py

```diff
@@ -0,0 +1,166 @@
+    async def handle_c2d(msg):
+        thief = json.loads(msg.data.decode()).get(Fields.THIEF, {})
+        logger.info("Received {}".format(thief))
+
+        if not thief:
+            logger.warning("No thief object in payload")
+            return
+
+        if thief.get(Fields.RUN_ID) != run_id:
+            logger.warning(
+                "run_id does not match: expected={}, received={}".format(
+                    run_id, thief.get(Fields.RUN_ID)
+                )
+            )
+            return
+
+        cmd = thief.get(Fields.CMD)
+        if cmd not in [
+            Commands.PAIR_RESPONSE,
+            Commands.OPERATION_RESPONSE,
+            Commands.METHOD_RESPONSE,
+            Commands.C2D_RESPONSE,
+        ]:
+            logger.warning("unknown cmd: {}".format(cmd))
+            return
+
+        operation_ids = thief.get(Fields.OPERATION_IDS, [])
+        if not operation_ids:
+            operation_ids = [
+                thief.get(Fields.OPERATION_ID),
+            ]
+
+        logger.info("Received {} message with {}".format(thief[Fields.CMD], operation_ids))
+
+        for operation_id in operation_ids:
+            running_op = running_operation_list.get(operation_id)
+            if running_op:
+                logger.info("setting event for message {}".format(operation_id))
+                running_op.result_message = msg
+                event_loop.call_soon_threadsafe(running_op.complete)
+            else:
+                logger.warning("Received unknown operationId: {}:".format(operation_id))
+
+    connected_client.on_message_received = handle_c2d
+
+
+@pytest.fixture(scope="class")
+async def paired_client(
+    connected_client, running_operation_list, c2d_waiter, run_id, requested_service_pool
+):
+    start_time = time.time()
+    while time.time() - start_time <= PAIRING_REQUEST_TIMEOUT_INTERVAL_IN_SECONDS:
+        running_op = running_operation_list.make_event_based_operation(event_module=asyncio)
+
+        pairing_payload = {
+            Fields.THIEF: {
+                Fields.CMD: Commands.PAIR_WITH_SERVICE_APP,
+                Fields.OPERATION_ID: running_op.id,
+                Fields.REQUESTED_SERVICE_POOL: requested_service_pool,
+            }
+        }
+        msg = create_message_from_dict(pairing_payload, None, run_id)
+        await connected_client.send_message(msg)
+
+        logger.info("Waiting for pairing response")
+        try:
+            await asyncio.wait_for(
+                running_op.event.wait(), PAIRING_REQUEST_SEND_INTERVAL_IN_SECONDS
+            )
+        except asyncio.TimeoutError:
+            pass
+        else:
+            logger.info("pairing response received")
+            msg = json.loads(running_op.result_message.data.decode())
+            return collections.namedtuple("ConnectedClient", "client service_instance_id")(
+                connected_client, msg[Fields.THIEF][Fields.SERVICE_INSTANCE_ID]
+            )
+
+    assert False
```

Comment:
If messages are supported for assertions, I think a message here would be useful

Reply:
done.  changed to an exception with a message.